Vizzuality/trase
# Fix/sankey card parameters

## Pivotal Tracker

https://www.pivotaltracker.com/story/show/170873735

## Description

This ended up being much bigger than initially suspected. The logic for keeping the stored links in sync with current node and node type identifiers (during a data update) was not reliable, and I replaced it with something simpler. However, it is still brittle, because of how coupled it is with front end parameter syntax.

To clean up the ActiveRecord model, I extracted the managament of query params to a separate service, which recognises the following types of params:
- array, e.g. `selectedNodesIds[]=x`
- single element, e.g. `extraColumnNodeId=x`
- comma separated list, e.g. `sources=x,y`
- comma separated list with "excluded" prefix (😨), e.g. `sources=excluded_x,y`
- the weirdest of all, custom columns param format, e.g. `selectedColumnsIds=0_1-1_2`

To store the link and return the query params to front end we don't need to look inside the values, and we return them as saved. Params are parsed during the data update process, for purposes of updating stale identifiers. Change the front end params format => this breaks.

This also includes a smaller requirement change of only allowing 3 cards for level 3.

## Testing instructions

Create a link, check clicking on the card opens the expected workspace.

* Also, running the data update process is an important check for this, I did that.
